include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -54,7 +54,7 @@ namespace aspect
          * beginning of the program. Checks preconditions.
          */
         void
-        initialize ();
+        initialize (const types::boundary_id boundary_id);
```

Comment:
while you're also modifying things here, could you mark this function as virtual?